aiida/backends/tests/engine/test_ports.py

```diff
@@ -98,3 +98,29 @@ def test_serialize_type_check(self):
         # The `assertRaisesRegexp` method is deprecated in python 3 but assertRaisesRegex` does not exist in python 2
         with self.assertRaisesRegexp(TypeError, '.*{}.*{}.*'.format(base_namespace, nested_namespace)):
             port_namespace.serialize({'some': {'nested': {'namespace': {Dict()}}}})
+
+    def test_lambda_default(self):
+        """Test that an input port can specify a lambda as a default."""
+        port_namespace = PortNamespace('base')
+
+        # Defining lambda for default that returns incorrect type should not except at construction
+        port_namespace['port'] = InputPort('port', valid_type=Int, default=lambda: 'string')
+
+        # However, pre processing the namespace, which shall evaluate the default and validate it will fail
```

Comment:
Would you mind checking this explicitly (that validation raises?)

Reply:
Done